api/app/src/main/kotlin/packit/service/PacketService.kt

```diff
@@ -117,6 +120,29 @@ class BasePacketService(
         return this.joinToString("") { "%02x".format(it) }
     }
 
+    override fun streamZip(paths: List<String>, id: String, output: OutputStream) {
+        val hashesByPath = getMetadataBy(id).files
+            .filter { it.path in paths }
+            .associateBy({ it.path }, { it.hash })
+
+        if (hashesByPath.size != paths.size) {
```

Comment:
Could throw a slightly friendlier error here including the filenames which weren't found - would just need to filter to the paths which weren't in metadata files mapped to their names.